src/main/java/de/unijena/cheminf/fragment/fingerprint/FragmentFingerprinter.java

```diff
@@ -0,0 +1,464 @@
+        }
+    }
+    //
+    /**
+     * Returns bit array for specified list.
+     *
+     * @param aListOfUniqueSmiles is a list that stores fragments in the form of unique SMILES.
+     * @return int[] bit array
+     * @throws NullPointerException is thrown if the list aListOfUniqueSmiles is null.
+     * @throws IllegalArgumentException is thrown if the list aListOfUniqueSmiles contains blank/empty strings.
+     */
+    public int[] getBitArray(List<String> aListOfUniqueSmiles) throws NullPointerException, IllegalArgumentException {
+        Objects.requireNonNull(aListOfUniqueSmiles, "aListOfUniqueSmiles (list of string instances) is null.");
+        for (String tmpUniqueSmiles : aListOfUniqueSmiles) {
+            Objects.requireNonNull(tmpUniqueSmiles, "aListOfUniqueSmiles (at least one list element) is null.");
+            if (tmpUniqueSmiles.isBlank() || tmpUniqueSmiles.isEmpty()) {
+                throw new IllegalArgumentException("aListOfUniqueSmiles (at least one list element) is blank/empty.");
+            }
+        }
+        Set<String> tmpSet = new HashSet<>(aListOfUniqueSmiles);
+        List<String> tmpListWithoutPossibleDuplicates = new ArrayList<>(tmpSet);
+        return this.createBitArray(tmpListWithoutPossibleDuplicates);
+    }
+    //
+    /**
+     * Returns bit array for the specified map (molecule).
+     * @see #getBitArray(List)
+     *
+     * @param aUniqueSmilesToFrequencyMap is a map that maps fragments in the form of unique SMILES to the
+     *                                    frequency of unique SMILES.
+     * @return int[] bit array
+     * @throws NullPointerException is thrown if the map aUniqueSmilesToFrequencyMap is
+     * null or contains keys or values that are null respectively.
+     * @throws IllegalArgumentException is thrown if the map aUniqueSmilesToFrequencyMap
+     * contains keys or values that are blank/empty, respectively.
+     */
+    public int[] getBitArray(Map<String,Integer> aUniqueSmilesToFrequencyMap) throws NullPointerException, IllegalArgumentException {
+        Objects.requireNonNull(aUniqueSmilesToFrequencyMap, "aUniqueSmilesToFrequencyMap (Map of string and integer instances) is null.");
+        List<String> tmpListOfUniqueSmiles = new ArrayList<>();
+        for(String tmpUniqueSmiles : aUniqueSmilesToFrequencyMap.keySet()) {
+            if(tmpUniqueSmiles == null || aUniqueSmilesToFrequencyMap.get(tmpUniqueSmiles) == null) {
+                throw new NullPointerException("aUniqueSmilesToFrequencyMap (Map of string and integer instances) contains " +
+                        "instances that are null.");
+            }
+            if(tmpUniqueSmiles.isBlank() || tmpUniqueSmiles.isEmpty()) {
+                throw new IllegalArgumentException("aUniqueSmilesToFrequencyMap (Map of strings an integer instances) contains strings that are blank/empty.");
+            }
+            tmpListOfUniqueSmiles.add(tmpUniqueSmiles);
+        }
```

Comment:
ArrayList has no initial capacity? I'm confused. See here: https://stackoverflow.com/questions/4172480/whats-meant-by-parameter-int-initial-capacity-in-an-arraylist

Reply:
I agree with Felix. But I think it must be the keys of the map that need to be given to the other method.

Also, this might be a bit misleading because the param is the map with frequencies but the return value is only bits and no counts. This should be stated explicitly in the doc, that this is just a convenience method and the given frequencies are not used.

Reply:
As you can see Jonas is right, this is misleading. As I did not recognize at first glance that it is the keys the map needed and not the values. But iirc we discussed this already